### ISSUE TITLE
508-defect-4 [SCREENREADER]: Semantic lists should only be used for lists

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#19618]
# [508-defect-4](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a screen reader user, I expect the announcement of a list to only be used for a list.

## Details

During empty or error states where no accounts exist, paragraphs of text are still written as ordered lists `ol` which is semantically incorrect and misleading to screen reader users. Screen readers will read out "list" before announcing one paragraph and a link.

## Acceptance Criteria

- [ ] Paragraphs for empty states are not wrapped by `ol` and `li`

## Environment

* Operating System: any
* Browser: all
* Screenreading device: any
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/profile/direct-deposit` in browser as user 226
2. Start screenreading device listed in Environment
3. Navigate to education benefits and VA payment history
4. Confirm both times the content is announced as a list

## Proposed Solution

Remove `ol` and `li` that wrap the paragraphs

## Screenshots or Trace Logs

<img width="1419" alt="Screen Shot 2021-02-08 at 4 48 25 PM" src="https://user-images.githubusercontent.com/14154792/107286200-42ff4e80-6a2e-11eb-85aa-d2894c90c857.png">

**Comment by mshea0606** [comment]
@erikphansen to follow up then provide an estimate

**Comment by Samara-Strauss** [comment]
I removed the direct deposit label and added the profile label since this pattern is used throughout the profile.

**Comment by erikphansen** [comment]
The fix for this is probably not very straightforward. I'm going to try to chat with Josh about it.

**Comment by joshkimux** [comment]
Erik and I synced up today and determined that we wouldn't be able to set clear acceptance criteria until after exploring all variations of the profile card(table?) component. We're aiming to:

- Take screenshots of all variations of the component (Erik)
- Document (a) the most appropriate semantic elements for each variation and (b) the semantic elements that work for each variation (Josh)
- Determine an implementation strategy to move forwards which will require moderate development work and direction from a designer ( @cgednalske 👋 😄 )

I've bumped this down to a defect-4 as this issue does not moderately hinder a Veteran from completing their task.

**Comment by joshkimux** [comment]
@MarciMcGuireGCIO Let me know if you'd prefer I edit this ticket to match the updated needs OR if I should close this and create a new one 🙇

**Comment by erikphansen** [comment]
I spoke with Josh about this earlier today. At present there isn't an easy solution to fix the markup for the offending component (the `ProfileInfoTable` https://github.com/department-of-veterans-affairs/vets-website/blob/1802128a2a965b862abb967df4de76ffd2d7c04d/src/applications/personalization/profile/components/ProfileInfoTable.jsx) because it's used in many different places in the Profile. Ideally the markup would be different depending on what data is being shown:

- In some contexts, such as service history, a `ol/li` might make sense.
- In some contexts, such as personal info, a `dl/dl` might make sense.
- In other contexts, list don't make sense and we might want to just use headings.

A full audit of all the places the component is used, along with recommendations for the ideal markup, might be the next step. Recommendations for changes to the component's markup would come out of that. The component might need to be updated so that it can conditionally change its underlying markup. Or, _maybe_ we could come up with consistent markup for the component that would be "good enough" to handle all of the ways the component is currently used.

Below are screenshots of all the places the component is used on the Profile.

![ProfileInfoTable - Account Security.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/3b9aed58-503b-4491-8657-72f0890102f5)
![ProfileInfoTable - Direct Deposit.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/1f070d86-1efc-4744-9443-76ce477756ba)
![ProfileInfoTable - Service History.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/0a4cee41-c8be-426e-931c-7c0841520567)
![ProfileInfoTable - Phone and Email.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/77fc1227-2f02-4479-a3a3-25eb2826048b)
![ProfileInfoTable - Personal Info and Addresses.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/cd91506a-3c6e-460e-84d1-3313d91d4898)

**Comment by erikphansen** [comment]
Sorry for my comment above repeating what Josh has already said. I blame Zenhub for showing me outdated content! Hopefully the screenshots will be helpful as we move forward.

**Comment by MarciMcGuireGCIO** [comment]
> @MarciMcGuireGCIO Let me know if you'd prefer I edit this ticket to match the updated needs OR if I should close this and create a new one 🙇

@joshkimux let's just keep this one so we have the comments for context.   Thanks!

**Comment by MarciMcGuireGCIO** [comment]
@erikphansen no worries!  Thanks for working with Josh on this!

**Comment by erikphansen** [comment]
@MarciMcGuireGCIO since the scope of this ticket needs to be reworked based on recommendations that are TBD, I'm going to move this back into the Backlog since it can't even be considered Ready to work on at this point. I'm also going to remove the estimate of 3 because we don't actually know what changes need to be made to the code.

**Comment by MarciMcGuireGCIO** [comment]
Thanks, @erikphansen!  Appreciate the update.

**Comment by MarciMcGuireGCIO** [comment]
Need a11y to provide recommendations on what markup should be.

**Comment by joshkimux** [comment]
# Recommendations for profile markup
Currently a WIP-- will be testing with Angela in the coming week. My current thoughts are as follows:
- Grey background with bitter font family should always be a heading
- For simple title description pairings, description lists work great e.g. service history
- Currently _uncertain_ if description lists will work well with more complex title description pairings e.g. email where there are buttons in one title/description pairing, a link in another title/description pairing, and also simple title/description pairings without interactive elements


## Account Security 

<details>
  <summary>Account security screenshot</summary>

![ProfileInfoTable - Account Security.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/3b9aed58-503b-4491-8657-72f0890102f5)

</details>

### Recommendation
[Pending review from Angela with this prototype on codepen](https://cdpn.io/joshkimux/debug/GRMWVoY/yokZEpZopEwA)

## Direct Deposit

<details>
  <summary>Direct deposit screenshot</summary>

![ProfileInfoTable - Direct Deposit.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/1f070d86-1efc-4744-9443-76ce477756ba)

</details>

### Recommendation
- Bitter font with the grey backgrounds are clearly semantically and visually headings.
- Sub title (bolded font e.g. Account) should be written as a description title `dt` within a description list `dl`
- All other cards _without_ the bolded subtitle should be written as paragraphs `p`

## Service History
<details>
  <summary>Service history screenshot</summary>

![ProfileInfoTable - Service History.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/0a4cee41-c8be-426e-931c-7c0841520567)

</details>

### Recommendation
- Bitter font with grey background should be a heading
- Service history and time descriptions should be written as a description list `dl`

## Phone and Email
<details>
  <summary>Phone and email screenshot</summary>

![ProfileInfoTable - Phone and Email.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/77fc1227-2f02-4479-a3a3-25eb2826048b)

</details>

### Recommendation
- Phone numbers shares same recommendation as service history, should be a heading with a description list `dl`
- Need to test with Angela to determine recommendation for emails

## Personal Information and Addresses
<details>
  <summary>Personal info and addresses screenshot</summary>

![ProfileInfoTable - Personal Info and Addresses.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/cd91506a-3c6e-460e-84d1-3313d91d4898)

</details>

### Recommendation
- Personal information shares same recommendation as phone numbers and service history (heading with `dl`)
- Need to test with Angela to determine recommendation for addresses

**Comment by MarciMcGuireGCIO** [comment]
Thank you @joshkimux and @AngelaFowler82

**Comment by joshkimux** [comment]
@MarciMcGuireGCIO we were able to conclude that it would be better to use an unordered list and headings pattern for the profile. I've documented the reasons above!

**Comment by MarciMcGuireGCIO** [comment]
Perfect!  This is super helpful!  Thanks again!

**Comment by SKasimow** [comment]
FYI - a ticket was opened for Discovery based on this thread #37041

**Comment by TiffanyPender** [comment]
With our recent full audit of Profile content (Personal Info, Contact Info, Direct Deposit Info and Notification Settings), I would also agree with the recommendation to utilize heading elements to organize content within each section. Very similar to the structure of "Notification Settings" where H3s are used for each "sub-category."

cc: @andaleliz @AngelaFowler82

**Comment by andaleliz** [comment]
@TiffanyPender thanks for taking a look at this! I'd like to confirm that for the rest of the content, you agree it should be an unordered list? [Here's the codepen example](https://codepen.io/joshkimux/pen/ZEXKzKx) we'd follow to restructure the page, per Josh's recommendation.

**Comment by joshkimux** [comment]
@andaleliz , I'm not too confident in that example, so please follow whatever recommendation @TiffanyPender provides as final 🙏 

To give more context here:
1. I think this page in isolation may be better without being nested in a list at all
2. However, I remember larger conversations about if we would follow a universal pattern that works for all profile pages (some pages semantically align with the need for lists more than others) or on a case by case basis

The latter point may influence if y'all choose to use unordered lists or not 😄

**Comment by andaleliz** [comment]
Oh, I see what you're saying @joshkimux. Thanks for the additional details there. We will take those points into consideration.

**Comment by Samara-Strauss** [comment]
Next steps for this work are broken down into the following tickets. When these tickets are completed/launched, we should be good to go to *finally* close this ticket:

- [ ] #53630
- [ ] 
#53631

- [ ] #53632

**Comment by joshkimux** [comment]
@adamwhitlock1 omg, I can always count on your team to knock out a11y tickets. Bless up. Big ups to @Samara-Strauss too, I see you prioritizing and knocking down these a11y tickets! Wish we had a bonus star system or something to shower on y'all for being such champs.

Tagging in @coforma-terry to find a CAIA team member to review ❤️

**Comment by mtcA6** [comment]
@joshkimux @coforma-terry hey all, checking in here, have you been able to review

**Comment by coforma-terry** [comment]
@Hey @mtcA6 - apologies I missed this one - I've gone ahead and tagged the right folks here so will get this reviewed asap for you - thanks!

@SarahKay8 - can you have a look at this one please :)
CC: @sara-amanda

**Comment by SarahKay8** [comment]
Hi @adamwhitlock1 where can I access your component in order to review it?

**Comment by SarahKay8** [comment]
Completed side by side review with Adam and validated profile-info-card component. Closing this out @coforma-terry @joshkimux
